server/matchmaker/matchmaker_queue.py

```diff
@@ -4,15 +4,33 @@
 from datetime import datetime, timezone
 from statistics import mean
 from typing import Deque, Dict
+import time
+import functools
 
 import server
+import server.metrics as metrics
 
 from ..decorators import with_logger
 from .algorithm import make_matches
 from .pop_timer import PopTimer
 from .search import Match, Search
 
 
+def timed_async_search(func):
+    @functools.wraps(func)
+    async def wrapper(*args, **kwargs):
+        queue_name = args[0].queue_name
+        metric = metrics.matchmaker_searches.labels(queue_name)
+        start_time = time.perf_counter()
```

Comment:
Not sure if `perf_counter` is the right function to use here, it's meant for "short" time durations whatever that means. Probably `time` or maybe even `monotonic` would be better since it's technically possible for players to be queued for hours: https://docs.python.org/3/library/time.html#time.monotonic